modin/experimental/backends/omnisci/query_compiler.py

```diff
@@ -142,7 +142,10 @@ def merge(self, right, **kwargs):
         left_index = kwargs.get("left_index", False)
         right_index = kwargs.get("right_index", False)
         """Only non-index joins with explicit 'on' are supported"""
-        if left_index is False and right_index is False and on is not None:
+        if left_index is False and right_index is False:
+            if on is None:
+                on = [c for c in self.columns if c in right.columns]
```

Comment:
```suggestion
        if left_index is False and right_index is False and on is None:
                on = self.columns.intersection(right.columns)
```

Reply:
You might have misread the code here. I add handling for `None` value of `on`. I don't want to handle `None` only and default to pandas for all other cases.

Reply:
oh, yeah, I misunderstood the indents)